File: ReactNativeApp/android/app/src/main/java/com/components/games/instances/juggling/JugglingState.java
package com.components.games.instances.juggling;

import com.ssvc.State;

public class JugglingState extends State {


    public final int BALL_COOL_DOWN = 0; // how many frames the ball should touch the groundline before we reset the jugglingcounter
    public final int FOOT_DETECT_RANGE = 15;
    public final int GROUND_LINE_TOUCHED_COOL_DOWN = 5;
    public final boolean stillJuggling = true;
    public double KNEE_LINE_ADJUSTMENT = 0.35;
    public int BALL_ON_GROUND_COUNTER = 1; // How many times we skip searching for a bounce after the ball has touched the ground
    public int determineBounceRange;

    //the lastBounceID is used to keep track of what the last analyzed frame is, to ensure we only count a juggle once
    public int lastBounceID = 0;
    public int determineBounceStartID = 0;
    public double groundLine;
    public double kneeLine;
    public double leftKneeLoc;
    public double rightKneeLoc;
    public double rightAnkleY, leftAnkleY; // for motion detection
    public boolean bounce = false;

    public int count = 0;


    @Override
    public String getDebugText() {
        return "JugglingState";
    }
}

File: ReactNativeApp/ios/Modules/SSVC/Games/Juggling/JugglingState.swift
//
//  JugglingState.swift
//  jogo
//
//  Created by arham on 18/11/2021.
//

import Foundation

class JugglingState: BaseGameState {
  
  internal static let BALLINVISIONRANGE: Int = 30
  internal final let KNEELINEADJUSTMENT: Double = 0.35
  
  internal final let BALLONGROUNDCOUNTER: Int = 5
  internal final let GROUNDLINETOUCHEDCOOLDOWN: Int = 5
  internal final let BALLCOOLDOWN: Int = 1
  internal final let FOOTDETECTRANGE: Int = 15
  internal final let BODYPARTRANGE: Int = 5
  
  internal var determineBounceRange: Int?
  internal var lastBounceId: Int = 0
  internal var determineBounceStartId: Int = 0
  
  internal var bounce: Bool = false
  internal var bounceDetectedLocation: DetectionLocation?
  internal var juggleLocation: DetectionLocation?
  
  internal var groundLine: Double?
  internal var kneeLine: Double?
  
  internal var leftKneeLoc: Double?
  internal var rightKneeLoc: Double?
  internal var leftHipLoc: Double?
  internal var rightHipLoc: Double?
  
  internal var rightAnkleY: Double?
  internal var leftAnkleY: Double?

  internal var ballLocations: [DetectionLocation]?
  internal var isJuggling: Bool = false
}
